tests/element_test_vectors.rs

```diff
@@ -163,17 +235,18 @@ trait ElementApi {
     /// This would have two groups, one with direct values that will be compared and another
     /// with embedded Ion text that will be parsed and compared.
     fn read_group<F1, F2>(
-        reader: Self::ElementReader,
+        reader: Self::ReaderApi,
         file_name: &str,
         value_assert: F1,
         group_assert: F2,
     ) -> IonResult<()>
     where
-        F1: Fn(&OwnedElement, &OwnedElement),
+        // group index, value 1 index, value 1, value 2 index, value 2
+        F1: Fn(usize, usize, &OwnedElement, usize, &OwnedElement),
```

Comment:
🗺️ This closure now takes indexes for the group being evaluated and for each of the elements being compared. This made debugging failures much easier.